toolchain/sem_ir/inst_namer.h

```diff
@@ -90,44 +95,51 @@ class InstNamer {
 
  private:
   // A space in which unique names can be allocated.
-  struct Namespace {
+  class Namespace {
+   private:
     // A result of a name lookup.
     struct NameResult;
 
+   public:
     // A name in a namespace, which might be redirected to refer to another name
     // for disambiguation purposes.
     class Name {
      public:
-      Name() : value_(nullptr) {}
-      explicit Name(llvm::StringMapIterator<NameResult> it) : value_(&*it) {}
+      explicit Name() : value_(nullptr) {}
+      explicit Name(llvm::StringMapIterator<NameResult> it,
+                    size_t base_name_size)
+          : value_(&*it), base_name_size_(base_name_size) {}
 
       explicit operator bool() const { return value_; }
 
-      auto str() const -> llvm::StringRef;
+      // Returns the disambiguated name.
+      auto GetDisambiguated() const -> llvm::StringRef;
```

Comment:
What do you think of calling this `GetFullName` or even `FullName` (and renaming `GetBase` to match)? This is called so often that it seems worth making it a little shorter and more noun-shaped.